internal/workflow/activities/unarchive.go

```diff
@@ -53,20 +54,35 @@ func (a *UnarchiveActivity) Execute(
 		"StripTopLevelDir", params.StripTopLevelDir,
 	)
 
-	u, err := unarchiver(params.SourcePath)
+	f, err := os.Open(params.SourcePath)
 	if err != nil {
-		return nil, fmt.Errorf("unarchive: unarchiver: %v", err)
+		return nil, fmt.Errorf("unarchive: open: %v", err)
 	}
-	if u == nil {
-		// Couldn't find an unarchiver, so this is probably a regular file.
-		// Return the source path unaltered, and IsDir as false.
-		a.logger.V(2).Info("Unarchive: not an archive, skipping.")
-		return &UnarchiveActivityResult{DestPath: params.SourcePath}, nil
+	defer f.Close()
+
+	r := io.Reader(f)
```

Comment:
I suspect this line is not needed, i.e. `*os.File` implements `io.Reader` and Go should be doing the conversion implicitly.

Reply:
You're right.  `archiver.Identify` returns an `io.Reader`, so I can't reassign the output to `f` like:
```go
format, f, err := archiver.Identify(params.SourcePath, f)
```

But I can just assign the output reader to a new var `r`:
```go
format, r, err := archiver.Identify(params.SourcePath, f)
```